src/pykeen/models/unimodal/triple_re.py

```diff
@@ -0,0 +1,113 @@
+        entity_normalizer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Hint[Initializer] = uniform_,
+        relation_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ) -> None:
+        r"""Initialize TripleRE via the :class:`pykeen.nn.modules.TripleREInteraction` interaction.
+
+        :param embedding_dim: The entity embedding dimension $d$.
+        :param p: The $l_p$ norm.
+        :param power_norm: Should the power norm be used?
+        :param entity_initializer: Entity initializer function. Defaults to :func:`torch.nn.init.uniform_`
+        :param entity_initializer_kwargs: Keyword arguments to be used when calling the entity initializer
+        :param entity_normalizer: Entity normalizer function. Defaults to :func:`torch.nn.functional.normalize`
+        :param entity_normalizer_kwargs: Keyword arguments to be used when calling the entity normalizer
+        :param relation_initializer: Relation initializer function. Defaults to :func:`torch.nn.init.uniform_`
+        :param relation_initializer_kwargs: Keyword arguments to be used when calling the relation initializer
+        :param kwargs: Remaining keyword arguments passed through to :class:`pykeen.models.ERModel`.
+        """
+        # TODO: the published code uses NodePiece representations instead
+        entity_normalizer_kwargs = _resolve_kwargs(
+            kwargs=entity_normalizer_kwargs,
+            default_kwargs=self.default_entity_normalizer_kwargs,
+        )
+        super().__init__(
+            interaction=TripleREInteraction,
+            interaction_kwargs=dict(p=p, power_norm=power_norm),
+            entity_representations=EmbeddingSpecification(
+                embedding_dim=embedding_dim,
+                initializer=entity_initializer,
+                initializer_kwargs=entity_initializer_kwargs,
+                normalizer=entity_normalizer,
+                normalizer_kwargs=entity_normalizer_kwargs,
+            ),
+            relation_representations=[
+                EmbeddingSpecification(
+                    embedding_dim=embedding_dim,
+                    initializer=relation_initializer,
+                    initializer_kwargs=relation_initializer_kwargs,
+                ),
+                EmbeddingSpecification(
+                    embedding_dim=embedding_dim,
+                    initializer=relation_initializer,
+                    initializer_kwargs=relation_initializer_kwargs,
+                ),
+                EmbeddingSpecification(
+                    embedding_dim=embedding_dim,
+                    initializer=relation_initializer,
+                    initializer_kwargs=relation_initializer_kwargs,
+                ),
+            ],
+            **kwargs,
+        )
+
+
+def _resolve_kwargs(kwargs: Optional[Mapping[str, Any]], default_kwargs: Mapping[str, Any]) -> Mapping[str, Any]:
+    # TODO: move to utils
```

Comment:
I think what you want here is a `collections.ChainMap`

Reply:
532a916